packages/ag-charts-community/src/util/memo.ts

```diff
@@ -12,3 +12,45 @@ export function memo<T, R>(params: T, fnGenerator: (params: T) => () => R): () =
 
     return memorizedFns.get(fnGenerator)?.get(serialisedParams) as () => R;
 }
+
+export function simpleMemorize<F extends (...args: any[]) => any>(fn: F) {
+    const primitveCache = new Map<string | number, {}>();
```

Comment:
I think this will lead to memory leaks - there's no way to free strings that get added

Reply:
Agreed in theory - practically though the number of different primitive values we're going to see is limited to theme names, and I think it adds a lot of complexity to callers to use object-only parameters, so I think this is ok for now.